src/test/java/org/assertj/core/internal/files/Files_assertHasBinaryContent_Test.java

```diff
@@ -66,19 +67,25 @@ void should_fail_if_actual_is_null() {
 
   @Test
   void should_fail_if_actual_is_not_file() {
+    // GIVEN
     AssertionInfo info = someInfo();
     File notAFile = new File("xyz");
-
+    // WHEN
     Throwable error = catchThrowable(() -> files.assertHasBinaryContent(info, notAFile, expected));
-
+    // THEN
     assertThat(error).isInstanceOf(AssertionError.class);
     verify(failures).failure(info, shouldBeFile(notAFile));
   }
 
   @Test
   void should_pass_if_file_has_expected_binary_content() throws IOException {
-    when(binaryDiff.diff(actual, expected)).thenReturn(BinaryDiffResult.noDiff());
-    files.assertHasBinaryContent(someInfo(), actual, expected);
+    File actual = newFile(tempDir.getAbsolutePath() + "/tmp.txt");
+    byte[] data = "actual".getBytes();
+    try (FileOutputStream myWriter = new FileOutputStream(actual)) {
+      myWriter.write(data, 0, data.length);
+    }
+    byte[] expected = "actual".getBytes();
+    unMockedFiles.assertHasBinaryContent(someInfo(), actual, expected);
```

Comment:
Take an approach similar to `Paths_assertHasBinaryContent_Test`, it's using a more concise way of writing files:
```
// call .toFile() other it returns a Path
Files.write(tempDir.resolve("actual"), "Content".getBytes()).toFile();
```

Reply:
This comment applies to other tests too (I mean looking at the equivalent test for Path)

Reply:
Since the two unresolved comments are on the same topic, I will ask my question here:

- We already have `import org.assertj.core.internal.Files;` so I have to use `java.nio.file.Files.write`. Any suggestions or comments?
- `tmpDir` is declared as type `protected File tempDir;` in FilesBaseTest. I could use `tempDir.toPath()` or change it to `protected Path tempDir;`. Or keep both types:
```
  @TempDir
  protected File tempDir;
  @TempDir
  protected Path tempDirPath;
```
Final:
```
  @Test
  void should_throw_error_if_actual_file_does_not_have_the_expected_size() throws IOException {
    // GIVEN
    File actual = java.nio.file.Files.write(tempDirPath.resolve("actual"), "content".getBytes()).toFile();
    // WHEN
    expectAssertionError(() -> files.assertHasSizeInBytes(INFO, actual, 36L));
    // THEN
    verify(failures).failure(INFO, shouldHaveSize(actual, 36L));
  }
```

Reply:
Also, should I replace usage of files from resources folder with the suggestion above?
```@BeforeAll
  static void setUpOnce() {
    actual = new File("src/test/resources/actual_file.txt");
  }
```

```@Test
  void should_pass_if_actual_has_expected_size() {
    files.assertHasSizeInBytes(INFO, actual, actual.length());
  }
```

Reply:
For files test let's use:
```
  @TempDir
  protected File tempDir;
```
unless I'm missing something we don't need a `Path tempDir` .

Reply:
we could add a helper method in `FilesBaseTest`  to wrap:
```java
java.nio.file.Files.write(tempDirPath.resolve("actual"), "content".getBytes()).toFile();
```
to something like:
```
writeFile("actual", "content");
```
That's a suggestion, maybe try it and see for yourself it it makes it easier otherwise don't worry too much ;-)